book/src/template_syntax.md

```diff
@@ -183,6 +183,21 @@ blocks from the base template with those from the child template. Inside
 a block in a child template, the `super()` macro can be called to render
 the parent block's contents.
 
+### Block Fragments
```

Comment:
I think it makes sense to mention the feature here since it relates to these blocks, but I feel like it makes sense to put the bulk of the documentation for this in `book/src/creating_templates.md` rather than here.